returnn/datasets/distrib_files.py

```diff
@@ -151,8 +153,27 @@ def __init__(
             distributed training scenaria
         :param _meta_info_cache: for internal use
         :param _distrib_info: for internal use
+        :param _num_shards: how many data shards there are
+        _param _shard_index: the index of the local shard
         """
-        super().__init__(**kwargs)
+        self.distrib_shard_files = distrib_shard_files
+        if distrib_shard_files:
+            assert _num_shards is None and _shard_index is None, (
```

Comment:
I think this check is wrong. When this is already some sub dataset (e.g. from DistribFilesDataset), this would not be None but 1/0.

Maybe move this code after the `super` and just check whether this is different from 1/0.